### PR TITLE
Improve logmes

### PR DESCRIPTION
**Pull request by mekolat**
[//]: # (**********************************)
[//]: # (** Fill in the following fields **)
[//]: # (**********************************)

[//]: # (Note: Lines beginning with syntax such as this one, are comments and will not be visible in your report!)

### Pull Request Prelude

[//]: # (Thank you for working on improving Hercules!)

[//]: # (Please complete these steps and check the following boxes by putting an `x` inside the brackets _before_ filing your Pull Request.)

- [X] I have followed [proper Hercules code styling][code].
- [X] I have read and understood the [contribution guidelines][cont] before making this PR.
- [X] I am aware that this PR will be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

Allows `logmes` to log to the atcommand table. This is also future-proof in case we want to add other log tables at some point

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

**Comment by HerculesWSAPI** [comment]
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/1843)

File: src/map/script.c [hunk 1]
```diff
@@ -15785,19 +15785,32 @@ BUILDIN(getmapxy)
...
+		type = script_getnum(st, 3);
+	}
+
+	switch (type) {
+	case LOGMES_ATCOMMAND:
+		logs->atcommand(sd, str);
+		break;
+	case LOGMES_NPC:
+		logs->npc(sd, str);
+	}
```
Comment (MishimaHaruna) [inline src/map/script.c]: How about adding a `default:` here that prints an error message? (so that a wrong value won't go silently unnoticed)

Comment (mekolat) [reply src/map/script.c]: good idea

**Comment by mekolat** [comment]
@MishimaHaruna done

**Comment by MishimaHaruna** [comment]
Reviewed 1 of 1 files at r1, 1 of 1 files at r2, 2 of 2 files at r3, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/herculesws/hercules/1843)*

**Pull request merged**